resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/FileUtil.java

```diff
@@ -339,8 +310,8 @@ public static IPath toPath(URI uri) {
 	public static final void transferStreams(InputStream source, OutputStream destination, String path,
 			IProgressMonitor monitor) throws CoreException {
 		SubMonitor subMonitor = SubMonitor.convert(monitor);
-		try {
-			try {
+		try (source) {
```

Comment:
No need for any actual change but FYI: `try (source; destination) {..}` would have reduced the nesting level by one.